File: shared/src/main/java/ru/privatenull/pnauth/message/MessageFormat.java
package ru.privatenull.pnauth.message;

import java.util.Locale;

public enum MessageFormat {
    LEGACY,
    MINI_MESSAGE,
    JSON,
    PLAIN;

    public static MessageFormat parse(String value) {
        if (value == null || value.isBlank()) return LEGACY;
        return switch (value.trim().toLowerCase(Locale.ROOT).replace('-', '_')) {
            case "legacy", "ampersand", "section" -> LEGACY;
            case "mini", "minimessage", "mini_message" -> MINI_MESSAGE;
            case "json", "component", "minecraft_json" -> JSON;
            case "plain", "text", "plain_text" -> PLAIN;
            default -> throw new IllegalArgumentException("Unknown message format: " + value);
        };
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/message/MessageFormat.kt
package ru.privatenull.pnauth.message

import java.util.Locale

enum class MessageFormat {
    LEGACY,
    MINI_MESSAGE,
    JSON,
    PLAIN;

    companion object {
        @JvmStatic
        fun parse(value: String?): MessageFormat {
            if (value.isNullOrBlank()) return LEGACY
            return when (value.trim().lowercase(Locale.ROOT).replace('-', '_')) {
                "legacy", "ampersand", "section" -> LEGACY
                "mini", "minimessage", "mini_message" -> MINI_MESSAGE
                "json", "component", "minecraft_json" -> JSON
                "plain", "text", "plain_text" -> PLAIN
                else -> throw IllegalArgumentException("Unknown message format: $value")
            }
        }
    }
}
